elk/training/train.py

```diff
@@ -152,37 +175,63 @@ def apply_to_layer(
                     {
                         **meta,
                         "ensembling": mode,
-                        **evaluate_preds(train_gt, train_credences, mode).to_dict(),
+                        **evaluate_preds(train.labels, train_credences, mode).to_dict(),
                         "train_loss": train_loss,
                     }
                 )
 
-                if val_lm_preds is not None:
+                if val.lm_preds is not None:
                     row_bufs["lm_eval"].append(
                         {
                             **meta,
                             "ensembling": mode,
-                            **evaluate_preds(val_gt, val_lm_preds, mode).to_dict(),
+                            **evaluate_preds(val.labels, val.lm_preds, mode).to_dict(),
                         }
                     )
 
-                if train_lm_preds is not None:
+                if train.lm_preds is not None:
                     row_bufs["train_lm_eval"].append(
                         {
                             **meta,
                             "ensembling": mode,
-                            **evaluate_preds(train_gt, train_lm_preds, mode).to_dict(),
-                        }
-                    )
-
-                for i, model in enumerate(lr_models):
-                    row_bufs["lr_eval"].append(
-                        {
-                            **meta,
-                            "ensembling": mode,
-                            "inlp_iter": i,
-                            **evaluate_preds(val_gt, model(val_h), mode).to_dict(),
+                            **evaluate_preds(
+                                train.labels, train.lm_preds, mode
+                            ).to_dict(),
                         }
                     )
 
-        return {k: pd.DataFrame(v) for k, v in row_bufs.items()}
+                if self.supervised != "none":
+                    if self.save_probs:
+                        out_probs[ds_name]["lr"][mode] = dict()
+
+                    for i, model in enumerate(lr_models):
+                        model.eval()
+                        val_lr_credences = model(val.hiddens)
+                        train_lr_credences = model(train.hiddens)
+                        if self.save_probs:
+                            out_probs[ds_name]["lr"][mode][i] = (
+                                get_logprobs(val_lr_credences, mode).detach().cpu()
+                            )
+
+                        row_bufs["train_lr_eval"].append(
```

Comment:
the names are getting a bit confusing. Maybe we should have a subfolder 
evals/
- trainset_lr.csv
- validationset_lr.csv
- trainset_ccs.csv
- etc.

Reply:
(Could be also included in a second pull-request)